src/iperf_api.c

```diff
@@ -1493,7 +1493,7 @@ iperf_parse_arguments(struct iperf_test *test, int argc, char **argv)
                 break;
             case 'O':
                 test->omit = atoi(optarg);
-                if (test->omit < 0 || test->omit > 60) {
+                if (test->omit < 0 || test->omit > MAX_TIME) {
```

Comment:
I think I've changed my mind on this. I like the ideas of getting rid of the hard-coded `60` and allowing something longer. I'm not sure that allowing up to a day of idle time, followed by up to a day of actual test, really has a practical use, and it ties up the server for longer than might be desired.

What about having another constant, say `MAX_OMIT_TIME`, defined to something like 600 (that's 10 minutes, which is a pretty long time), do the check against that?

Again, if anyone feels a need for a longer omit time, they hopefully have the technical skill to check out the source code, make the change, and compile it.

What do you think?

Reply:
The reason I used the `MAX_TIME` is because one of the use-cases for longer omit time mentioned in #1770 is "**get output only from the end of iperf connection**".  If this use-case is regarded as useful, then the maximum omit time should be the same as the maximum test time.

Another option for the above use-case, is to allow **negative** omit time, to mean that the omit time is for the test time **except** for the positive value of the "omit" time.  E.g. `-t 100 -O -20`  means omitting the first 80 seconds of the test.  In this case smaller (e.g. 600) absolute omit time is probably sufficient.

If the above use-case doesn't seem useful enough, or the negative value option is taken, then I agree that (absolute) 600 seconds should be sufficient.  In any case, I assume that defining `MAX_OMIT_TIME` can be useful for easier modifications by anyone that wants to change the limit.

Reply:
My understanding of how `--omit` works (which admittedly I didn't write myself) is a little different. If the user specifies `--omit`, iperf3 runs a "pre-test" for that number of seconds before it starts the actual test of however many seconds, blocks, or bytes. So, if I start up a client with `--omit 5 --time 20`, the client sends data for 5 seconds (which presumably is enough to overcome startup transients) and then does a 20-second test. The server knows to throw away those first 5 seconds and only report on the 20 seconds.

If someone wanted to only get data "from the end", they might do something like "--omit 15 --time 5`, which does a 15-second pre-test followed by 5 seconds of actual test. You could also think of this as saving only the last 5 seconds of a 20 second test.

My interpretation of #1770 is just that they needed an omit time longer than 60 seconds to set up some stuff for whatever they were doing. 60 seconds is a extremely long time by networking standards (in terms of RTT, buffer utilization, etc.) and I can't remember us at ESnet needing an omit time that long for anything we were doing. But I figure that supporting a slightly longer omit time can't hurt.

I understand that the semantics of `--omit` can be a little confusing. If I were designing the feature I might have done this a little differently.

Reply:
You are right.  I knew that the omit time is a pre-test period but somehow forgot that ...

OK. I changed the code as requested - limit omit time to `MAX_OMIT_TIME` which is set to 600.

Reply:
Thanks! Hoping to finish this off later this week.